src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -439,19 +444,19 @@ public static FormulaValue Split(IRContext irContext, StringValue[] args)
         private static FormulaValue Substitute(IRContext irContext, FormulaValue[] args)
         {
             var source = (StringValue)args[0];
+            var match = (StringValue)args[1];
+            var replacement = (StringValue)args[2];
 
-            if (args[1] is BlankValue || (args[1] is StringValue sv && string.IsNullOrEmpty(sv.Value)))
+            if (string.IsNullOrEmpty(match.Value))
             {
                 return source;
             }
 
-            var match = (StringValue)args[1];
-            var replacement = (StringValue)args[2];
-
             var instanceNum = -1;
-            if (args[3] is NumberValue nv)
+            if (args.Length > 3)
```

Comment:
Probably not a frequently hit scenario but should we short circuit and return source if args.Length <= 3 and (match == replacement )?

Reply:
This is a corner case that is already handled by the existing logic, so I'd rather have less code than more for a (negligible) performance improvement.